src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_gemv.cpp

```diff
@@ -0,0 +1,247 @@
+
+    if (!fc_params.compressed) {
+        return false;
+    }
+    const size_t scale_group_size = weights.IFM().v / fc_params.decompression_scale.Feature().v;
+    if (scale_group_size == 0 || scale_group_size % 16 != 0) {
+        return false;
+    }
+
+    // Data type re-check: only support f16:int4:f16
+    if (input.GetDType() != Datatype::F16 || output.GetDType() != Datatype::F16 ||
+        (weights.GetDType() != WeightsType::INT4 && weights.GetDType() != WeightsType::UINT4)) {
+        return false;
+    }
+
+    // Only support vector data as input, the data size should be aligned by 16 elements
+    auto input_size = get_input_bf_size(fc_params);
+    if (input_size.first > 1 || input_size.second == 0 || input_size.second % 16 != 0 || weights.IFM().v % 16 != 0) {
+        return false;
+    }
+
+    auto wl = weights.GetLayout();
+    auto wo = weights.OFM().v;
+
+    auto& fc_input = fc_params.inputs[0];
+    if (is_swiglu_fused(fc_params)) {
+        return false;
+    }
+
+    if (input_size.first != 0 && fc_input.is_dynamic()) {
+        if (input_size.first != 1) {
+            return false;
+        }
+        if (!(wl == WeightsLayout::os_is_yx_osv32_isv2 && wo % 32 == 0) &&
+            !(wl == WeightsLayout::os_is_yx_osv64_isv2 && wo % 64 == 0) &&
+            !(wl == WeightsLayout::os_iyx_osv16 && wo % 16 == 0)) {
+            return false;
+        }
+    }
+
+    if (input.GetLayout() == DataLayout::bfyx) {
+        // Padding on input is not supported.
+        if (input.X().pad.Total() != 0)
+            return false;
+        if (input.Y().pad.Total() != 0)
+            return false;
+    }
+
+    // We don't support 4d output
+    if (fc_params.outputs[0].GetLayout() == DataLayout::bfyx && fc_params.outputs[0].X().v > 1)
+        return false;
+
+    return true;
+}
+
+FullyConnected_GEMV::DispatchData FullyConnected_GEMV::SetDefault(const fully_connected_params& params,
+                                                                  int,
+                                                                  int /*kernel_number*/) const {
+    auto dispatchData = Parent::SetDefault(params);
+
+    std::vector<size_t> global = {params.weights.OFM().v, 1, 16};
+    if (params.weights.GetLayout() == WeightsLayout::os_iyx_osv16) {
+        global[0] = params.weights.OFM().v;
+    } else if (params.weights.GetLayout() == WeightsLayout::os_is_yx_osv32_isv2) {
+        global[0] = params.weights.OFM().v / 2;
+        // if (is_swiglu_fused(params)) {
+        //     global[0] = params.weights.OFM().v / 4;
+        // }
+    } else if (params.weights.GetLayout() == WeightsLayout::os_is_yx_osv64_isv2) {
+        global[0] = params.weights.OFM().v / 4;
+    }
+
+    dispatchData.gws = global;
+    dispatchData.lws = {16, 1, 16};
+
+    return dispatchData;
+}
+
+KernelsPriority FullyConnected_GEMV::GetKernelsPriority(const Params& /*params*/) const {
+    return FORCE_PRIORITY_9;
+}
```

Comment:
I think we can try to avoid `single_batch_optimized` logic by assigning higher kernel priority for static gemv kernel like this:
    return params.is_shape_agnostic ? FORCE_PRIORITY_9 : FORCE_PRIORITY_2;
```suggestion
KernelsPriority FullyConnected_GEMV::GetKernelsPriority(const Params& /*params*/) const {
    return params.is_shape_agnostic ? FORCE_PRIORITY_9 : FORCE_PRIORITY_2;
}
```